components/tor/tor_file_watcher.h

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_TOR_TOR_FILE_WATCHER_H_
+#define BRAVE_COMPONENTS_TOR_TOR_FILE_WATCHER_H_
+
+#include <memory>
+#include <vector>
+
+#include "base/callback.h"
+#include "base/files/file_path.h"
+#include "base/files/file_path_watcher.h"
+#include "base/memory/weak_ptr.h"
+
+namespace base {
+class SequencedTaskRunner;
+struct OnTaskRunnerDeleter;
+}  // namespace base
+
+namespace tor {
+
+FORWARD_DECLARE_TEST(TorFileWatcherTest, EatControlCookie);
+FORWARD_DECLARE_TEST(TorFileWatcherTest, EatControlPort);
+
+class TorFileWatcher {
```

Comment:
it would be better to document here that instances of this class self delete vs https://github.com/brave/brave-core/pull/7927/files#diff-e76f0bbcbb600ac467d8422bf9133cedae918a8dce6945e6cf916f35f4e36db8R105

Reply:
fixed in 730f9927c864fb2d71e5f9089fc107f7b351fc8b